src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -242,28 +419,55 @@ public long getArrivalTime() {
         return transaction.bytes;
     }
 
+    /**@return The transaction {@link Hash} identifier*/
     public Hash getHash() {
         return hash;
     }
 
+    /**
+     * This is a getter for the {@link AddressViewModel} associated with this {@link Transaction}.
+     *
+     * @param tangle The tangle reference for the database.
+     * @return The {@link AddressViewModel} of the {@link Transaction}.
+     * @throws Exception If the address cannot be found in the database, an exception is thrown.
+     */
     public AddressViewModel getAddress(Tangle tangle) throws Exception {
         if(address == null) {
             address = AddressViewModel.load(tangle, getAddressHash());
         }
         return address;
     }
 
+    /**
+     * This is a getter for the {@link TagViewModel} associated with this {@link Transaction}.
+     *
+     * @param tangle The tangle reference for the database.
+     * @return The {@link TagViewModel} of the {@link Transaction}.
+     * @throws Exception If the address cannot be found in the database, an exception is thrown.
+     */
     public TagViewModel getTag(Tangle tangle) throws Exception {
         return TagViewModel.load(tangle, getTagValue());
     }
 
+    /**
+     * This is a getter for the {@link AddressHash} identifier of a {@link Transaction}. If the {@link Hash} is null, a
+     * new one is created from the stored {@link #trits}.
+     *
+     * @return The {@link AddressHash} identifier.
+     */
     public Hash getAddressHash() {
         if(transaction.address == null) {
             transaction.address = HashFactory.ADDRESS.create(trits(), ADDRESS_TRINARY_OFFSET);
         }
         return transaction.address;
     }
 
+    /**
+     * This is a getter for the {@link ObsoleteTagHash} identifier of a {@link Transaction}. If the {@link Hash} is
```

Comment:
> If the {@link Hash} is null, a new one is created from the stored

it's better to mention this in only 1 place - no need to re-iterate it (it's an implementation detail - lazy evaluation).